src/host/plan-object.js

```diff
@@ -0,0 +1,177 @@
+ * @param {*} log
+ * @returns {AriaATCIHost.TestPlan}
+ */
+export function addLogToTestPlan(testPlan, log) {
+  return { ...testPlan, log: [...testPlan.log, log] };
+}
+
+/**
+ * @param {AriaATCIHost.TestPlan} testPlan
+ * @param {{filepath: string}} testFilepath
+ * @returns {AriaATCIHost.TestPlan}
+ */
+export function addTestLogToTestPlan(testPlan, { filepath: testFilepath }) {
+  const test = testPlan.tests.find(({ filepath }) => filepath === testFilepath);
+  return {
+    ...testPlan,
+    tests: arrayUtil.replace(testPlan.tests, test, {
+      ...test,
+      log: [...test.log, testPlan.log.length - 1],
+    }),
+  };
+}
+
+/**
+ * @param {AriaATCIHost.TestPlan} testPlan
+ * @param {string} testFilepath
+ * @param {*} result
+ * @param {AriaATCIHost.TestPlan}
+ */
+export function addTestResultToTestPlan(testPlan, testFilepath, result) {
+  const test = testPlan.tests.find(({ filepath }) => filepath === testFilepath);
+  return {
+    ...testPlan,
+    tests: arrayUtil.replace(testPlan.tests, test, { ...test, results: [...test.results, result] }),
+  };
+}
+
+/**
+ * @param {*} serverOptions
+ * @returns {TestPlanServerOptionsPartial}
+ */
+function validateTestPlanServerOptionsPartial(serverOptions) {
+  invariant(typeof serverOptions === 'object' && serverOptions !== null);
+  for (const key of Object.keys(serverOptions)) {
+    invariant(['baseUrl', 'files'].includes(key));
+  }
+  if (serverOptions.baseUrl) {
+    validateTestPlanURL(serverOptions.baseUrl);
+  }
+  if (serverOptions.files) {
+    invariant(
+      typeof serverOptions.files === 'object' &&
+        serverOptions.files !== null &&
+        Array.isArray(serverOptions.files)
+    );
+    for (const file of serverOptions.files) {
+      invariant(typeof file === 'string');
+    }
+  }
+  return serverOptions;
+}
+
+/**
+ * @param {*} url
+ * @returns {AriaATCIShared.BaseURL}
+ */
+function validateTestPlanURL(url) {
+  invariant(typeof url === 'object' && url !== null);
+  for (const key of Object.keys(url)) {
+    invariant(['protocol', 'hostname', 'port', 'pathname'].includes(key));
+  }
+  invariant(typeof url.protocol === 'string');
+  invariant(typeof url.hostname === 'string');
+  validatePort(url.port);
+  invariant(typeof url.pathname === 'string');
+  return url;
+}
+
+function validatePort(port) {
+  invariant(typeof port === 'number', () => `typeof ${port} === 'number'`);
+  invariant(port > 0 && port < 0x10000);
+  return port;
+}
+
+/**
+ * @param {*} file
+ * @returns {TestPlanFile}
+ */
+function validateTestPlanFile(file) {
+  invariant(typeof file === 'object' && file !== null);
+  for (const key of Object.keys(file)) {
+    invariant(['name', 'buffer'].includes(key));
+  }
+  invariant(typeof file.name === 'string');
+  validateUint8Array(file.buffer);
+  return file;
+}
+
+function validateUint8Array(buffer) {
+  invariant(
+    buffer instanceof Uint8Array,
+    () => `'${buffer && buffer.constructor.name}' === 'Uint8Array'`
```

Comment:
> The difference between those two definitions is most apparent in code like this (which reads, "transform the buffer into a buffer"), and I think it may make the code harder to follow elsewhere.

I replaced the `Buffer.from(...).toString()` with a `TextDecoder.decode`.

> A name like bufferData would help distinguish between the Uint8Array which stores a portion of memory and the Buffer instance.